core/domain/story_domain.py

```diff
@@ -555,7 +628,30 @@ def get_node_index(self, node_id):
                 return ind
         return None
 
-    def get_ordered_nodes(self):
+    def strict_get_node_index(self, node_id: str) -> int:
```

Comment:
Thinking about this, should we follow similar places in the codebase where we add `strict` argument and overload the type for it?

Reply:
Done!